kliff/transforms/parameter_transforms.py

```diff
@@ -0,0 +1,62 @@
+from typing import TYPE_CHECKING, Dict, List, Union
+
+import numpy as np
+
+if TYPE_CHECKING:
+    from kliff.models.parameter import Parameter
+
+
+class ParameterTransform:
+    """
+    Abstract class to transform parameters and inverse transform it back.
+
+    Subclass can implement
+        - transform
+        - inverse
+    """
+
+    def __init__(self, name):
+        self.name = name
+
+    def transform(
+        self, model_param: Union["Parameter", np.ndarray]
```

Comment:
If I do that then we will run into the issue of circular imports. As ParameterTransforms uses Parameter for type annotation, but Parameter will also use ParameterTransforms.
Example Line 52 in parameter.py
```
    def __new__(
        cls,
        input_array: Union[np.ndarray, float, int],
        name: str = None,
        transform: "ParameterTransform" = None,
        bounds: np.ndarray = None,
        is_mutable: bool = False,
        index: int = None,
        opt_mask: np.ndarray = None,
    ):
```

Reply:
As per pep standards, `TYPE_CHECKING` is always false during interpreter execution. So I dont think it will be an issue. Adding these to lines just help the static analyzers to get appropriate modules for type checking (they match the string annotations and use the appropriate classes for analyzing attributes etc). Can remove if needed. It will not affect the functionality in any way.